### PR TITLE
Configurable Lock Prefix

### PR DESCRIPTION
**Pull request by harshanarayana**
Fix the current implementation which uses the hardcoded "/locks" prefix and makes it use a configurable prefix that can be passed via the ETCD client. This behavior is consistent with the ETCD2 implementation as well.

### CODE REVIEW

**Comment by harshanarayana** [comment]
This PR will address the changes required in #353

**Comment by harshanarayana** [comment]
@kragniz Would you be able to take a look at this PR and see if it looks good ?

File: README.rst [hunk 1]
```diff
@@ -50,6 +50,10 @@ Basic usage:
     with etcd.lock('doot-machine') as lock:
         # do something
 
+    # locks with prefix
+    etcd.lock_prefix = "/locks"
+    lock = etcd.lock('thing')   # Lock will be generated under /locks/thing
```
Comment (kragniz) [inline README.rst]: I'm not sure about this API. I'd rather `etcd.lock_prefix` wasn't directly used as a properly, instead you use `etcd.lock('thing', prefix='/locks')`, or `etcd3.client(lock_prefix='/locks')` only

Comment (harshanarayana) [reply README.rst]: Makes sense. Changing it accordingly.

**Comment by harshanarayana** [comment]
@kragniz I think something went wrong with the Travis build. Terminated the build with 

```bash
tests/test_etcd3.py .......
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
```

File: etcd3/client.py [hunk 2]
```diff
@@ -177,10 +179,14 @@ def __enter__(self):
 
     def __exit__(self, *args):
         self.close()
-        
+
     def __del__(self):
         self.close()
 
+    @property
+    def lock_prefix(self):
```
Comment (kragniz) [inline etcd3/client.py]: can you remove this `@property` too please

Comment (harshanarayana) [reply etcd3/client.py]: This was added to enable accessing the `_lock_prefix` private variable of `Etcd3Client`. Should I make that a public variable instead in that case?

**Comment by harshanarayana** [comment]
@kragniz Also, another quick question. Since the current implementation was using `/locks` as the lock prefix, should I default the Lock prefix to use the same so that It will have a backward compatibility for the current usage and won't break anything?

**Pull request closed without merging**